javascript/ql/src/semmle/javascript/dataflow/TypeInference.qll

```diff
@@ -92,9 +84,6 @@ class AnalyzedNode extends DataFlow::Node {
     exists(DataFlow::Incompleteness cause |
       isIncomplete(cause) and result = TIndefiniteAbstractValue(cause)
     )
-    or
-    result = globalFlowPred().getALocalValue() and
-    shouldTrackGlobally(result)
```

Comment:
I can see that this is now unnecessary for call-graph building, but won't it still affect type inference results for cases where we reference, but don't call, a function?

Reply:
That's a good point. It's one the reasons I included https://github.com/Semmle/ql/pull/1707/commits/d60467ef819a1cdec1e23fb4655d751b8910d131 to use global access paths for `.prototype` detection in `FunctionStyleClass`.

Maybe the better way is to change `SourceNode.getAFunctionValue()` to look through global access paths?